### PR TITLE
Adds thieves to the game

### PR DESCRIPTION
**Pull request by jcpirtle**

### CODE REVIEW

**Comment by Willamin** [comment]
@jcpirtle Just play testing this, it seems like after a thief stops grabbing me they die, even without me attacking them. Is that intended behavior?

**Comment by jcpirtle** [comment]
yes, once the adventurer struggles away the thief is supposed to die

File: source/index.js [hunk 1]
```diff
@@ -36,17 +36,12 @@ var state = {
         wave: {
             capacity: 4,
             monsters: [
+
                 MONSTERS.RED_SLIME,
                 MONSTERS.BLUE_SLIME,
-                MONSTERS.RED_ORC,
```
Comment (Willamin) [inline source/index.js:41]: The other monsters should probably be added back in before merging this.

**Comment by jcpirtle** [comment]
ok, i added the rest of the monsters back

**Review by Willamin — APPROVED** [review]
LGTM!

File: source/scripts/data/monsters.js [hunk 2]
```diff
@@ -147,5 +147,34 @@ export default {
...
+        grabCounter: function() {
+            this.turnCount = this.turnCount || 0
+            //the adventurer cannot be grabbed by another monster and he cannot grab him more than once
+            if(this.game.adventurer.grabCount == 0 && this.turnCount == 0) {
+                this.turnCount = this.turnCount + 1
+                this.game.adventurer.grabCount += 3
+                this.game.adventurer.grabMonster = this
+            }
+        }
+    },
```
Comment (ehgoodenough) [inline source/scripts/data/monsters.js:179]: Is there anyway we could refactor out grabCounter, so both thieves can share it? The only difference is the grabCount, which can just as well be replaced with the monster's health variable.

File: source/scripts/tests/ThievesTest.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    game.monsters[0].onAction()
+    game.adventurer.move({y: +1})
+
+    Expect(game.monsters[0]).to.eql(undefined)
+    Expect(game.adventurer.position).to.eql({x: 3, y: 3})
+
+    game.adventurer.move({y: +1})
+
+    Expect(game.monsters[0]).to.eql(undefined)
+    Expect(game.adventurer.position).to.eql({x: 3, y: 4})
```
Comment (ehgoodenough) [inline source/scripts/tests/ThievesTest.js:55]: Good tests. :]

**Review by ehgoodenough — APPROVED** [review]
Other than some code duplication, looks good to me. Let's get this merged. Could I get a playable build?

**Comment by jcpirtle** [comment]
https://mocsarcade.github.io/enchiridion/thieves

**Pull request merged**